### PR TITLE
updating documentation for u256

### PR DESCRIPTION
**Pull request by xunilrj**
## Description

This PR is part of https://github.com/FuelLabs/sway/pull/4794 and updates documentation for `u256` and improves test for edge cases.

Code also prints more info when tests run in `verbose` mode.

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by anton-trunov — APPROVED** [review]

File: docs/book/src/reference/solidity_differences.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ The underlying virtual machine targeted by Sway is the FuelVM, specified [here](
 
 ## Word Size
 
-Words in the FuelVM are 64 bits (8 bytes), rather than the 256 bits (32 bytes) of the EVM. Therefore, primitive integers only go up to `u64`, and hashes (the `b256` type) are not in registers but rather in memory. A `b256` is therefore a pointer to a 32-byte memory region containing the hash value.
+Words in the FuelVM are 64 bits (8 bytes), rather than the 256 bits (32 bytes) of the EVM. Therefore, all primitive integers smaller and including `u64` are stored in registers; and hashes (the `b256` type) are not stored in registers but rather in memory. A `b256` is therefore a pointer to a 32-byte memory region containing the hash value.
```
[the remark this answers is not in the document]
Comment (IGI-111) [reply docs/book/src/reference/solidity_differences.md:11]: I don't really like that the documentation is singling out the FuelVM implementation as I'd like us to have a more formal distinction between Sway and the VM, but for the benefits of clarity, we should probably keep this and mention `u256` as well.

[the remark this answers is not in the document]
Comment (xunilrj) [reply docs/book/src/reference/solidity_differences.md]: Done.

**Review by anton-trunov — APPROVED** [review]

**Review by sdankel — APPROVED** [review]
